### PR TITLE
[#165028951] Update italia ts commons v5.1.1

### PR DESCRIPTION
**Pull request by Undermaken**
This PR aims to upgrade italia-ts-commons to version 5.1.1
The main benefit of this upgrade is a better errors handling on network requests. In this way either user and dev should know more about errors details

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#165028951](https://www.pivotaltracker.com/story/show/165028951): Aggiornare italia-ts-commons con il supporto agli errori di validazione delle richieste

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cloudify** [comment]
This pull request **introduces 2 alerts** when merging 15b01e78e9e81b81a0673225a6542e72c5ff11a1 into fd84fbcb2ae15cb80712899786a62467629719e7 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-9bb9f871617b7bb3792aa0c9a4e2389eee545256)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: ts/sagas/startup/loadServiceRequestHandler.ts [hunk 1]
```diff
@@ -23,8 +22,12 @@ export function* loadServiceRequestHandler(
       { service_id: action.payload }
     );
 
-    if (response !== undefined && response.status === 200) {
-      yield put(loadService.success(response.value));
+    if (response.isLeft()) {
+      throw readableReport(response.value);
```
Comment (gunzip) [inline ts/sagas/startup/loadServiceRequestHandler.ts]:
```suggestion
      throw Error(readableReport(response.value));
```

File: ts/sagas/messages/messages.ts [hunk 2]
```diff
@@ -61,26 +61,26 @@ export function* loadMessage(
...
 ): IterableIterator<Effect | Either<Error, MessageWithContentPO>> {
   try {
     const response: SagaCallReturnType<typeof getMessage> = yield call(
       getMessage,
       { id: meta.id }
     );
-
-    if (response === undefined || response.status !== 200) {
+    if (response.isLeft()) {
+      throw readableReport(response.value);
```
Comment (gunzip) [inline ts/sagas/messages/messages.ts]:
```suggestion
      throw Error(readableReport(response.value));
```

**Comment by cloudify** [comment]
This pull request **introduces 1 alert** when merging 2b9c480fe4b24c7f3c94822b7234c5fb12599aa1 into 9af86171e2826d8a6ecdaa85333c6e63ad0f1e26 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-3719b6f00d45db054a0c20d72010a3b3638b2294)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by cloudify** [comment]
This pull request **introduces 1 alert** when merging 97cdbd802658a252e2ac67151a5c173f30a3959d into e237d910982b619e483bc2a7ce70b6275d3d53a5 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-4d7725646747301de1c6a94529bb4577cd9b4910)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: ts/api/backend.ts [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
   createFetchRequestForApi,
   ioResponseDecoder as ioD,
   IPostApiRequestType,
   IResponseType,
+  RequestHeaderProducer,
+  RequestHeaders,
   ResponseDecoder
 } from "italia-ts-commons/lib/requests";
-
+import { Omit } from "italia-ts-commons/lib/types";
```
Comment (cloudify) [inline ts/api/backend.ts]:
```suggestion
import { Omit } from "italia-ts-commons/lib/types";

```

File: ts/components/messages/MessageListItemComponent.tsx [hunk 4]
```diff
@@ -179,6 +178,7 @@ export class MessageListItemComponent extends React.Component<Props> {
     // value if createdAt cannot be converted to a Date
     // TODO: get created_at from CreatedMessageWithoutContent to avoid waiting
     //       for the message to load
+    /*
```
Comment (cloudify) [inline ts/components/messages/MessageListItemComponent.tsx]: why is this commented out?

File: ts/components/messages/MessageListItemComponent.tsx [hunk 5]
```diff
@@ -238,7 +239,7 @@ export class MessageListItemComponent extends React.Component<Props> {
               <Col size={10}>{serviceComponent}</Col>
               <Col size={3}>
                 <Text style={styles.dateText} rightAlign={true}>
-                  {uiCreatedAt}
+                  {""}
```
Comment (cloudify) [inline ts/components/messages/MessageListItemComponent.tsx]: ?

File: ts/sagas/backendInfo.ts [hunk 6]
```diff
@@ -20,9 +20,12 @@ const BACKEND_INFO_LOAD_INTERVAL = 60 * 60 * 1000;
 const BACKEND_INFO_RETRY_INTERVAL = 60 * 60 * 10 * 1000;
 
 function* backendInfoWatcher(): IterableIterator<Effect> {
+  console.warn(apiUrlPrefix);
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: !

File: ts/sagas/backendInfo.ts [hunk 7]
```diff
@@ -20,9 +20,12 @@ const BACKEND_INFO_LOAD_INTERVAL = 60 * 60 * 1000;
...
   const backendPublicClient = BackendPublicClient(apiUrlPrefix);
 
-  function getServerInfo(): Promise<BasicResponseType<ServerInfo> | undefined> {
+  function getServerInfo(): Promise<
+    t.Validation<BasicResponseType<ServerInfo> | undefined>
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: the new client will never return `undefined` (`undefined` gets replaced by `Left<Errors, T>`)

File: ts/sagas/backendInfo.ts [hunk 8]
```diff
@@ -34,13 +37,16 @@ function* backendInfoWatcher(): IterableIterator<Effect> {
...
-
-    if (backendInfoResponse && backendInfoResponse.status === 200) {
-      yield put(backendInfoLoadSuccess(backendInfoResponse.value));
-      setInstabugUserAttribute(
-        "backendVersion",
-        backendInfoResponse.value.version
-      );
+    console.warn(apiUrlPrefix);
+    if (
+      backendInfoResponse !== undefined &&
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: it's never `undefined`

File: ts/sagas/backendInfo.ts [hunk 9]
```diff
@@ -34,13 +37,16 @@ function* backendInfoWatcher(): IterableIterator<Effect> {
...
+      backendInfoResponse.isRight() &&
+      backendInfoResponse.value &&
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: `backendInfoResponse.value` is always defined

**Comment by cloudify** [comment]
This pull request **introduces 2 alerts** when merging dad2846a29d741359ddf1f768f0c08f5bfd1ea00 into 17680b228ae377d50edee2d076f9bdab6d85adb5 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-e960f28da334a3ba233e16514b09916af18043a8)

**new alerts:**

* 1 for Unreachable statement
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by cloudify** [comment]
This pull request **introduces 1 alert** when merging 56b6243f99ae512e6008546700de7a5c8073271e into 17680b228ae377d50edee2d076f9bdab6d85adb5 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-b983d2d3abc36f986c93556a3800cc2f47a5ab34)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by cloudify** [comment]
@Undermaken @fpersico what's the status of this PR? it's been opened almost 20 days ago, please focus on merging it today, thx

**Comment by Undermaken** [comment]
@cloudify @fpersico This PR is ready for review. As requested in the last sprint update planning I keep it updated and ready to merge. However I kept it draft because **io-utils** generates a wrong types for pagoPa models: in _requestsTypes.ts_ (definitions/pagopa) it uses **string** as a type instead of **t.string** as I reported few times (https://www.pivotaltracker.com/story/show/165028951/comments/202260962). This is because with this version of italia-ts-commons the gen command for pagopa api definitions must be 

`rimraf definitions/pagopa && mkdir -p definitions/pagopa && gen-api-models --api-spec https://wisp2.pagopa.gov.it/pp-restapi-CD/v2/api-docs --out-dir ./definitions/pagopa --no-strict --request-types --response-decoders --default-error-type string`

changing the error type from **--default-error-type Error** to **--default-error-type string**

**Comment by cloudify** [comment]
@Undermaken yesterday I checked the code of `gen-api-models` and the behavior is actually correct

File: package.json [hunk 10]
```diff
@@ -12,8 +12,8 @@
     "tsc:noemit": "tsc --noemit",
     "lint": "tslint -p . -c tslint.json -t verbose",
     "lint-autofix": "tslint -p . -c tslint.json -t verbose --fix",
+    "generate:pagopa-api-definitions": "rimraf definitions/pagopa && mkdir -p definitions/pagopa && gen-api-models --api-spec https://wisp2.pagopa.gov.it/pp-restapi-CD/v2/api-docs --out-dir ./definitions/pagopa --no-strict --request-types --response-decoders --default-error-type string",
```
Comment (cloudify) [inline package.json]: `string` is not a type


```suggestion
    "generate:pagopa-api-definitions": "rimraf definitions/pagopa && mkdir -p definitions/pagopa && gen-api-models --api-spec https://wisp2.pagopa.gov.it/pp-restapi-CD/v2/api-docs --out-dir ./definitions/pagopa --no-strict --request-types --response-decoders --default-error-type String",
```

Comment (Undermaken) [reply package.json]: [UPDATE] io-utils needs a fix about using default types for error responses (when not specified) https://github.com/teamdigitale/io-utils/issues/101

**Comment by cloudify** [comment]
🎉

**Pull request merged**